fleetspeak/src/client/daemonservice/execution/execution.go

```diff
@@ -167,8 +168,9 @@ func New(daemonServiceName string, cfg *dspb.Config, sc service.Context) (*Execu
 		outData: &dspb.StdOutputData{},
 		lastOut: time.Now(),
 
-		dead:        make(chan struct{}),
-		startupData: make(chan *fcpb.StartupData, 1),
+		dead:              make(chan struct{}),
+		startupData:       make(chan *fcpb.StartupData, 1),
+		waitForGoroutines: func() { wg.Wait() },
```

Comment:
could this be `waitForGoroutines: wg.Wait`? Or does `wg` need to be accessed at the time of the function call instead of the time of creating the Execution object

Reply:
Yes, done.